libraries/core_libs/network/include/network/tarcap/taraxa_capability.hpp

```diff
@@ -35,18 +35,25 @@ class TaraxaPeer;
 class TaraxaCapability : public dev::p2p::CapabilityFace {
  public:
   TaraxaCapability(std::weak_ptr<dev::p2p::Host> host, const dev::KeyPair &key, const NetworkConfig &conf,
-                   std::shared_ptr<DbStorage> db = {}, std::shared_ptr<PbftManager> pbft_mgr = {},
-                   std::shared_ptr<PbftChain> pbft_chain = {}, std::shared_ptr<VoteManager> vote_mgr = {},
-                   std::shared_ptr<NextVotesManager> next_votes_mgr = {}, std::shared_ptr<DagManager> dag_mgr = {},
-                   std::shared_ptr<DagBlockManager> dag_blk_mgr = {}, std::shared_ptr<TransactionManager> trx_mgr = {},
-                   addr_t const &node_addr = {});
+                   unsigned version);
 
   virtual ~TaraxaCapability() = default;
   TaraxaCapability(const TaraxaCapability &ro) = delete;
   TaraxaCapability &operator=(const TaraxaCapability &ro) = delete;
   TaraxaCapability(TaraxaCapability &&ro) = delete;
   TaraxaCapability &operator=(TaraxaCapability &&ro) = delete;
-
+  static std::shared_ptr<TaraxaCapability> make(
```

Comment:
isn't it better to be called make_shared ? Just a question

Reply:
Host and some other classes have the same `make` naming. I don't think it is clearifies a lot, but we can rename all of them

Reply:
I will leave up to you